lib/iris/fileformats/netcdf.py

```diff
@@ -1958,7 +1949,11 @@ def set_packing_ncattrs(cfvar):
             # Explicitly assign the fill_value, which will be the type default
             # in the case of an unmasked array.
             if packing is None:
-                fill_value = cube.fill_value
+                if hasattr(cube.lazy_data(), 'fill_value'):
+                    fill_value = cube.lazy_data().fill_value
+                else:
+                    fill_value = None
+
```

Comment:
@lbdreyer We can't get the `fill_value` if the data is lazy, so that changes the logic here some what ...

For lines 1951-1957, I propose the following:
```python
if packing is None:
    fill_value = None
    if not cube.has_lazy_data() and ma.isMaskedArray(cube.data):
        fill_value = cube.data.fill_value
    dtype = cube.dtype.newbyteorder('=')
```